test/spark/test_spark_getting.py

```diff
@@ -79,6 +80,51 @@ def test_getitem_list_array(sc):
     b = array(x, sc, axis=(0, 1))
     assert allclose(b[rows, cols, dept].toarray(), x[rows, cols, dept])
 
+def test_getitem_mixed(sc):
+
+    x = arange(4*4*4*4).reshape(4, 4, 4, 4)
+    b = array(x, sc, axis=(0, 1))
+
+    i = [0, 1]
+    s = slice(1, 3)
+    assert allclose(b[i, :, :, :].toarray(), x[i, :, :, :])
+    assert allclose(b[i, s, s, s].toarray(), x[i, s, s, s])
+    assert allclose(b[:, :, i, :].toarray(), x[:, :, i, :])
+    assert allclose(b[s, s, i, s].toarray(), x[s, s, i, s])
+
+    i = [1]
+    assert allclose(b[i, :, :, :].toarray(), x[i, :, :, :])
+    assert allclose(b[:, :, i, :].toarray(), x[:, :, i, :])
+
+    i = [[0, 1], [1, 0]]
+    with pytest.raises(ValueError):
+        b[i, :, :, :]
+
+def test_bounds(sc):
+
+    x = arange(5)
+    b = array(x, sc)
+
+    # out of bounds
```

Comment:
Completely agree -- it would have helped me solve any number of off-by-one errors way more quickly. It does allow for some hacks, but a clean API should need hacks on strange edge-cases!